src/main/java/org/opentripplanner/raptor/rangeraptor/context/SearchContext.java

```diff
@@ -254,10 +255,29 @@ private static <T extends RaptorTripSchedule> RaptorTransitCalculator<T> createC
     RaptorRequest<T> r,
     RaptorTuningParameters t
   ) {
+    var forward = r.searchDirection().isForward();
     SearchParams s = r.searchParams();
-    return r.searchDirection().isForward()
-      ? new ForwardRaptorTransitCalculator<>(s, t)
-      : new ReverseRaptorTransitCalculator<>(s, t);
+
+    if (r.multiCriteria().transitPassthroughRequest().isPresent()) {
+      var requiredC2 = r
+        .multiCriteria()
+        .transitPassthroughRequest()
+        .get()
+        .passthroughPoints()
+        .size();
+      if (forward) {
+        return new PassThroughForwardRaptorTransitCalculator<>(s, t, requiredC2);
+      } else {
+        // TODO: 2023-06-29 via pass through: we should have pass-through calculator for reverse as well
+        return new ReverseRaptorTransitCalculator<>(s, t);
```

Comment:
We need PassThrough implementation for this. Otherwise "arriveBy" searches won't work.